### ISSUE TITLE
Hi how do you process your input data, and if i want to trainning other data , this mode is ok for me ?

### ISSUE THREAD

**Issue by fuyazhou** [kwonmha/Improving-RNN-recommendation-model#3]
Hi how do you process your input data, and if i want to trainning other data , this mode is ok for me ? Thanks!

**Comment by kwonmha** [comment]
It's OK.
If you want to use other dataset, **use preprocess.py** to process the dataset into suitable format and store them in particular directory.
preprocess.py assumes that the format of dataset is "userID movieID rating timestamp(optional)".
It collects every movie each users watched and make lists like
user1 movie1 rating1 movie2 rating2 ...
user2 movie10 rating10 movie11 rating11 ...
... 
Then, you can use them to train and test.

**Comment by fuyazhou** [comment]
Thanks very much!
Your code little difficult for me, I have some other questions, all these questions LSTM
1, What is the time length of the input?
2, How many LSTM layer stack together? and how many neurons?
3, What is your specific network structure?   for example， input_layer ( ? * ? ) --> Embedding layer( ? * ?)---> LSTM layer( ? * ? ) -->Dropout layer ---> Dense layer---> output
4, What is the output format?
5, If the input does not have rating, just have user_id and movie_id, it is ok to train?
6, Do you have code only about LSTM, can you send to me?   Email: fuyazhou80@gmail.com
Thanks!!

**Comment by fuyazhou** [comment]
hi, are you still here?

**Comment by kwonmha** [comment]
1. maximum input length is set to 30 in default. You can modify with --max_length
2. You can set the number of stack with --r_l. For example, 100-50 value means 2 stacked layers, and the dimension of first lstm is 100 and that of second lstm is 59.
3. Input layer - Embedding layer - lstm layer - dense layer - output.
There's no dropout layer.
Each input is one-hot form. Which means the size of an input vector is the number of items in dataset.
It looks like i1 = [0, 0, 1, 0, 0 .....] 
And total input will be like i1, i2, i3... which mean the movies an user watched.
You can set the size of embedding vector by --r_emb.
4. The output format is a vector each dimension of which means how possible corresponding movie will be watched next. Its size is the number of movies. The 10 movies that have highest possibility score will be recommended.
5. Its ok.
6. I can't understand the meaning of 'code only about LSTM' in your question. See recurrent_layers.py for codes related to tensorflow LSTM. If you want to know about raw lstm codes implemented in numpy or codes of lstm in tensorflow, you'd better use google.

**Comment by fuyazhou** [comment]
thanks very much
